storage/operation/approvals.go

```diff
@@ -28,7 +28,7 @@ func RetrieveResultApproval(r storage.Reader, approvalID flow.Identifier, approv
 	return RetrieveByKey(r, MakePrefix(codeResultApproval, approvalID), approval)
 }
 
-// UnsafeIndexResultApproval inserts a ResultApproval ID keyed by ExecutionResult ID
+// IndexResultApproval inserts a ResultApproval ID keyed by ExecutionResult ID
```

Comment:
I am objecting to removing the unsafe qualifier. We know from experience, that devs don't read the documentation in detail. The unsafe qualifier is our last line of defence.
This is not mitigated by the lock, because the method still doesn't check